tests/unit/test_resources.py

```diff
@@ -34,7 +36,13 @@ class StatMock(object):
 
 
 def test_external(mocker):
+    assert "dash_core_components" in ComponentRegistry.registry
+    assert "dash_html_components" in ComponentRegistry.registry
+    mocker.patch("dash.development.base_component.ComponentRegistry.registry")
+    ComponentRegistry.registry = {"dash_core_components", "dash_html_components"}
```

Comment:
These tests would fail previously if run in tandem with the integration tests, because they import and register various other component packages. After this change, I can simply call `pytest` locally to run all the Python-based tests in this repo.

There's still one that fails locally for me: The iframe sandbox test [rdif001 fails on its last line](https://github.com/plotly/dash/blob/cd2ff7c25c459d86bd0f60a578f89910425b94e1/tests/integration/renderer/test_iframe.py#L60) - it can't see the log error it's supposed to have. I'm not sure what's going on here but it seems we've run afoul of some sort of security restriction: in the Chrome devtools I can't even see the DOM inside the iframe (though I can interact with this DOM via Selenium).
<img width="457" alt="Screen Shot 2020-12-16 at 9 42 59 PM" src="https://user-images.githubusercontent.com/2678795/102437201-29222c00-3fe8-11eb-9df9-2ddf7103517b.png">
Anyway this test still runs fine on CI, but if what I'm seeing is new behavior in the latest Chrome (v87) we may need to sort this out soon.

Reply:
Running this locally on Chrome 87, getting:
![image](https://user-images.githubusercontent.com/8092993/102510137-49d49b00-4055-11eb-87fd-0fff7e33ac0e.png)

Running this locally on Chrome 89 (Canary), getting:
![image](https://user-images.githubusercontent.com/8092993/102510422-9ae48f00-4055-11eb-94db-42b9d5b16807.png)

Moving from warning to warning+error.

Changing the iframe sandbox options to
```
<iframe src="{0}" sandbox="allow-scripts allow-same-origin">
```

- fixes the problem on Chrome 87
- fixes the warning on Chrome 89 but the error remains
![image](https://user-images.githubusercontent.com/8092993/102510590-c8c9d380-4055-11eb-945a-e95c17dfccac.png)

Attempting to find additional information.

Reply:
Trying to access`window.localStorage` from the console from Chrome 89 provides a slightly better explanation
![image](https://user-images.githubusercontent.com/8092993/102511618-fe22f100-4056-11eb-9715-5aeb808c02b3.png)
-- this is new

Reply:
In both Chrome 87 and 89 `localStorage` is not accessible from data URL but in 89 it's shown from the start. In both 87/89 running `window.localStorage` from the console gives
![image](https://user-images.githubusercontent.com/8092993/102511879-4e9a4e80-4057-11eb-8def-1e40493835f6.png)

Reply:
To be clear, the test _requires_ that a warning is logged so we're not trying to "fix" all the warnings. I'm not sure it's really important that we test for these logs, the main thing is that the app still works and doesn't error out trying to access cookies. So the easy answer here would be to simply remove the log check. But the "full" solution I guess would be to serve a real container page with an iframe in it, rather than a `data:` url, and keep the check as is.

For reference this feature was introduced in https://github.com/plotly/dash/pull/1080